vector/src/main/java/im/vector/app/features/onboarding/OnboardingViewModel.kt

```diff
@@ -289,9 +290,19 @@ class OnboardingViewModel @AssistedInject constructor(
     }
 
     private fun emitFlowResultViewEvent(flowResult: FlowResult) {
-        _viewEvents.post(OnboardingViewEvents.RegistrationFlowResult(flowResult, isRegistrationStarted))
+        withState { state ->
+            val orderedResult = when {
+                state.hasSelectedMatrixOrg() && vectorFeatures.isOnboardingCombinedRegisterEnabled() -> flowResult.overrideOrder()
+                else                                                                                 -> flowResult
+            }
+            _viewEvents.post(OnboardingViewEvents.RegistrationFlowResult(orderedResult, isRegistrationStarted))
+        }
     }
 
+    private fun OnboardingViewState.hasSelectedMatrixOrg() = selectedHomeserver.userFacingUrl == matrixOrgUrl
+
+    private fun FlowResult.overrideOrder() = copy(missingStages = missingStages.sortedWith(FtueMissingRegistrationStagesComparator()))
```

Comment:
It's strange that this extension is actually doing a `copy`of the state. I would keep the `copy` at the call site.

Reply:
will inline :+1: